maas/data_source_maas_boot_source_selection.go

```diff
@@ -0,0 +1,118 @@
+package maas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canonical/gomaasclient/client"
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceMaasBootSourceSelection() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMaasBootSourceSelectionRead,
+		Description: "Provides a resource to fetch a MAAS boot source selection.",
+
+		Schema: map[string]*schema.Schema{
+			"arches": {
+				Type:        schema.TypeSet,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Optional:    true,
+				Description: "The architecture list for this resource",
+			},
+			"boot_source_id": {
```

Comment:
Please rename to `boot_source` and in the description please use  `the boot source database ID` or something similar to inform the practitioner about the field